src/ParallelAlgorithms/NonlinearSolver/NewtonRaphson/ElementActions.hpp

```diff
@@ -0,0 +1,508 @@
+
+  template <typename DbTagsList, typename... InboxTags, typename Metavariables,
+            typename ArrayIndex, typename ActionList,
+            typename ParallelComponent>
+  static std::tuple<db::DataBox<DbTagsList>&&> apply(
+      db::DataBox<DbTagsList>& box,
+      const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+      const Parallel::GlobalCache<Metavariables>& /*cache*/,
+      const ArrayIndex& array_index, const ActionList /*meta*/,
+      const ParallelComponent* const /*meta*/) noexcept {
+    if (UNLIKELY(get<logging::Tags::Verbosity<OptionsGroup>>(box) >=
+                 ::Verbosity::Debug)) {
+      Parallel::printf(
+          "%s " + Options::name<OptionsGroup>() + "(%zu): Prepare step\n",
+          array_index,
+          db::get<Convergence::Tags::IterationId<OptionsGroup>>(box) + 1);
+    }
+
+    db::mutate<Convergence::Tags::IterationId<OptionsGroup>, correction_tag,
+               linear_operator_applied_to_correction_tag,
+               NonlinearSolver::Tags::Globalization<
+                   Convergence::Tags::IterationId<OptionsGroup>>,
+               NonlinearSolver::Tags::StepLength<OptionsGroup>,
+               globalization_fields_tag>(
+        make_not_null(&box),
+        [](const gsl::not_null<size_t*> iteration_id, const auto correction,
+           const auto linear_operator_applied_to_correction,
+           const gsl::not_null<size_t*> globalization_iteration_id,
+           const gsl::not_null<double*> step_length,
+           const auto globalization_fields, const auto& fields,
+           const double damping_factor) noexcept {
+          ++(*iteration_id);
+          // Begin the linear solve with a zero initial guess
+          *correction =
+              make_with_value<typename correction_tag::type>(fields, 0.);
+          // Since the initial guess is zero, we don't need to apply the linear
+          // operator to it but can just set it to zero as well. Linear things
+          // are nice :)
+          *linear_operator_applied_to_correction = make_with_value<
+              typename linear_operator_applied_to_correction_tag::type>(fields,
+                                                                        0.);
+          // Prepare line search globalization
+          *globalization_iteration_id = 0;
+          *step_length = damping_factor;
+          *globalization_fields = fields;
+        },
+        db::get<fields_tag>(box),
+        db::get<NonlinearSolver::Tags::DampingFactor<OptionsGroup>>(box));
+    return {std::move(box)};
+  }
+};
+
+// Between `PrepareStep` and this action the linear solver has run, so the
+// `correction_tag` is now filled with the solution to the linearized problem.
+// We just take a step in the direction of the correction.
+//
+// The `Globalize` action will jump back here in case the step turned out to be
+// insufficient. It will have adjusted the step length, so we can just try again
```

Comment:
Here I read `insufficient` as too small, but isn't it usually that the step length is decreased by globalization? Maybe something like `out to not decrease the residual sufficiently` (if this is true :P )